WebInstall/explainer.md

```diff
@@ -1,10 +1,374 @@
+3. UA tries to fetch the background document present at the `<install_url>` and its manifest file.
+4. If fetched document has a manifest file, continue. Else reject with `DataError`.
+5. If manifest file linked to the fetched document has an `id` field defined, continue. Else reject with `DataError`.
+6. UA shows the acquisition/installation confirmation UX (prompt/dialog). If the user accepts, continue. Else reject with `AbortError`. 
+7. Promise resolves with processed `id` of installed app and application follows the platform's post-install UX (adds to Dock/opens app/adds to mobile app drawer).
+
+#### Background Document (2 param)
+
+1. User gesture activates code that calls the `install(<install_url>, <manifest_id>)` method. 
+2. If the `<install_url>` is not the current document, the UA asks for permission to perform installations (if not previously granted). Else reject with `AbortError`. 
+3. UA tries to fetch the background document present at the `<install_url>` and its manifest file.
+4. If fetched document has a manifest file, continue. Else reject with `DataError`.
+5. If `<manifest_id>` matches the processed `id` from the manifest of the fetched document, continue. Else reject with `DataError`.
+6. UA shows the acquisition confirmation UX (prompt/dialog). If the user accepts, continue. Else reject with `AbortError`.
+7. Promise resolves with processed `id` of installed app and application follows the platform's post-install UX (adds to Dock/opens app/adds to mobile app drawer).
+
+### Sample code
+
+```javascript
+/* tries to install the current document */
+const installApp = async () => {
+    if (!navigator.install) return; // api not supported
+    try {
+        await navigator.install();
+    } catch(err) {
+        switch(err.name){
+            case 'AbortError':
+                /* Operation was aborted*/
+                break;
+        }
+    }
+};
+```
+
+```javascript
+/* tries to install a background document (app) */
+
+const installApp = async (install_url, manifest_id) => {
+    if (!navigator.install) return; // api not supported
+    try {
+        await navigator.install(install_url, manifest_id);
+    } catch(err) {
+        switch(err.name){
+            case 'AbortError':
+                /* Operation was aborted*/
+                break;
+            case 'DataError':
+                /*issue with manifest file or id*/
+                break;
+        }
+    }
+};
+```
+
+### Installing an *already* installed application
+
+In the case that the `navigator.install` method is invoked to install an application that is already installed in the device, it is up to the UA to decide the relevant default behaviour. For example, the UA can choose to open (or ask to open) the app.  
+* The promise will resolve if the application opens.
+* The promise rejects otherwise, with an `AbortError` or `DataError` depending on the UA's implementation.
```

Comment:
I think it's problematic to say that the choice of error in this case is left up to the UA, as it then diminishes the value to developers that there are two different errors.  They can no longer rely on their difference in meaning if that's the case.

Reply:
👍🏼 AbortError should be the option then for when an app is not opened (if there is a prompt).